### ISSUE TITLE
Not showing Achievement Points when first load.

### ISSUE THREAD

**Issue by seeson** [TheKrowi/Krowi_AchievementFilter#92]
Error showing AP and Calendar.
![image](https://user-images.githubusercontent.com/12587456/205981714-61c66f47-12f0-4913-bde3-97d4f026d5aa.png)
```
1x Interface\AddOns\Krowi_AchievementFilter\API.lua:133: attempt to index local 'category' (a nil value)
[string "@Interface\AddOns\Krowi_AchievementFilter\API.lua"]:133: in function `KrowiAF_SelectCategory'
[string "@Interface\AddOns\Krowi_AchievementFilter\GUI\Misc\AlertFrame\AlertFrame.lua"]:13: in function <...AchievementFilter\GUI\Misc\AlertFrame\AlertFrame.lua:4>
[string "=[C]"]: ?

Locals:
category = nil
collapsed = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'category' (a nil value)"
addon = <table> {
 ReplaceVarsWithReloadReq = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:523
 WorldMapButtons = <table> {
 }
 GUI = <table> {
 }
 SearchOptions = <table> {
 }
 Filters = <table> {
 }
 ExcludeAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:210
 Data = <table> {
 }
 IsCustomModifierKeyDown = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:649
 GetCategoryInfoTitle = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:22
 IsShadowlandsRetail = false
 Objects = <table> {
 }
 DelayObjects = <table> {
 }
 GetAchievementNumbers = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:51
 DelayFunction = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:622
 BuildCache = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:319
 Options = <table> {
 }
 Diagnostics = <table> {
 }
 TooltipData = <table> {
 }
 EventData = <table> {
 }
 GlobalStrings = <table> {
 }
 ChangeAchievementMicroButtonOnClick = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:603
 GetUsableSets = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:573
 Modifiers = <table> {
 }
 GetAchievementInfo = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:532
 GetNextAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:543
 ExtraWrathStrings = <table> {
 }
 Colors = <table> {
 }
 GetInstanceInfoName = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:10
 GetMapName = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:38
 GetSecondsSince = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:513
 Util = <table> {
 }
 MetaData = <table> {
 }
 GetFirstAchievementId = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:3
 MakeWindowMovable = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:506
 WatchAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:160
 InGuildView = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:12
 TrackingAchievements = <table> {
 }
 GetAchievementsInZone = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:20
 MakeWindowStatic = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:469
 GetVariantSetIDs = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:585
 GetLFGDungeonInfo = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:34
 ResetCache = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:349
 OverwriteFunctions = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:369
 OnAchievementEarned = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:353
 Plugins = <table> {
 }
 L = <table> {
 }
 Icon = <table> {
 }
 Credits = <table> {
 }
 GetActiveCovenant = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:16
 HookAchievementFrameOnShow = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:441
 Bindings = <table> {
 }
 IncludeAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:187
 TabsOrder = <table> {
 }
 AddToTrackingAchievementsCategories = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:174
 Event = <table> {
 }
 ClearWatchAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:136
 IsDragonflightRetail = false
 Tutorials = <table
```
```
4x ...chievementFilter\GUI\Misc\AchievementFrameHeader.lua:185: attempt to concatenate field 'Realm' (a nil value)
[string "@Interface\AddOns\Krowi_AchievementFilter\GUI\Misc\AchievementFrameHeader.lua"]:185: in function <...chievementFilter\GUI\Misc\AchievementFrameHeader.lua:176>

Locals:
self = <unnamed> {
 0 = <userdata>
}
characters = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
 4 = <table> {
 }
 5 = <table> {
 }
 6 = <table> {
 }
}
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
 4 = <table> {
 }
 5 = <table> {
 }
 6 = <table> {
 }
}
(for control) = 3
_ = 3
character = <table> {
 Guid = "Player-4818-00B78093"
 Name = "Kardashian"
 Faction = "Alliance"
 Class = "PRIEST"
 Points = 0
}
r = 1
g = 1
b = 1
name = "Kardashian"
(*temporary) = "Kardashian"
(*temporary) = " - "
(*temporary) = nil
(*temporary) = "Kardashian"
(*temporary) = "艾隆纳亚"
(*temporary) = "Kardashian"
(*temporary) = "艾隆纳亚"
(*temporary) = 1
(*temporary) = 1
(*temporary) = 1
(*temporary) = "attempt to concatenate field 'Realm' (a nil value)"
addon = <table> {
 ReplaceVarsWithReloadReq = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:523
 WorldMapButtons = <table> {
 }
 GUI = <table> {
 }
 SearchOptions = <table> {
 }
 Filters = <table> {
 }
 ExcludeAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:210
 Data = <table> {
 }
 IsCustomModifierKeyDown = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:649
 GetCategoryInfoTitle = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:22
 IsShadowlandsRetail = false
 Objects = <table> {
 }
 DelayObjects = <table> {
 }
 GetAchievementNumbers = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:51
 DelayFunction = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:622
 BuildCache = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:319
 Options = <table> {
 }
 Diagnostics = <table> {
 }
 TooltipData = <table> {
 }
 EventData = <table> {
 }
 GlobalStrings = <table> {
 }
 ChangeAchievementMicroButtonOnClick = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:603
 GetUsableSets = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:573
 Modifiers = <table> {
 }
 GetAchievementInfo = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:532
 GetNextAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:543
 ExtraWrathStrings = <table> {
 }
 Colors = <table> {
 }
 GetInstanceInfoName = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:10
 GetMapName = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:38
 GetSecondsSince = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:513
 Util = <table> {
 }
 MetaData = <table> {
 }
 GetFirstAchievementId = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:3
 MakeWindowMovable = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:506
 WatchAchievement = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:160
 InGuildView = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:12
 TrackingAchievements = <table> {
 }
 GetAchievementsInZone = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:20
 MakeWindowStatic = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:469
 GetVariantSetIDs = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:585
 GetLFGDungeonInfo = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Localization\GlobalStrings.lua:34
 ResetCache = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:349
 OverwriteFunctions = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:369
 OnAchievementEarned = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:353
 Plugins = <table> {
 }
 L = <table> {
 }
 Icon = <table> {
 }
 Credits = <table> {
 }
 GetActiveCovenant = <function> defined @Interface\AddOns\Krowi_AchievementFilter\Globals.lua:16
 HookAchi
```

**Comment by TheKrowi** [comment]
Hey, atm I'm not able to reproduce this behavior. Can you please write down the steps I should take to trigger it?
Providing me your savedvariables might also be useful.

**Comment by seeson** [comment]
Hi there, nothing special, it happens every time I log into the character for the first time from character vault.
[Krowi_AchievementFilter.zip](https://github.com/TheKrowi/Krowi_AchievementFilter/files/10191512/Krowi_AchievementFilter.zip)

**Comment by TheKrowi** [comment]
Do you by any chance have TacoTip installed? Another user identified that addon to interfere with KAF.

Secondly, if you do not use TacoTip, can you try and find which addon is interefering on your end? Disabling/enabling the top/bottom half of your addons and repeating this, should give you the culprit in the least amount of reloads.

**Comment by TheKrowi** [comment]
https://github.com/anzz1/TacoTip/issues/23

**Comment by seeson** [comment]
Ahhhh, I do have TacoTip installed. kk, so what is the solution?

**Comment by anzz1** [comment]
@seeson We are working on a solution to fix the conflict, so expect an update to TacoTip or Krowi Achievement Filter (or both) soon.

**Comment by anzz1** [comment]
Fixed in TacoTip release [0.3.5](https://github.com/anzz1/TacoTip/releases/tag/0.3.5)
